### DISCUSSION TITLE
Intro to programming

### DISCUSSION THREAD

**Discussion by jakegros** [USU-ITLS-x265-23-summer/class-repo#62] [Week 2 - Discussions]
I am way more familiar with html than Javascript. I found the forum was helpful for one thing I was stuck on, and to learn more of things I was curious about.  The html project-based learning was helpful and seeing the progress was great. The project being broken down into bite sized chunks can be really really helpful when learning. On the other hand, it can feel too spoonfed for some folks.   Feedback was helpful in general but sometimes useless when missing something tedious, hence utilizing the forum. 
For Javascript, I only had a very limited experience with an intro to [drawing/animation from Khan Academy](https://www.khanacademy.org/computing/computer-programming/programming). (This is not at all necessary for this class, just including it for those interested now that I've mentioned it).
I found the JS projects to be way more intensive but also more of a rewarding project. I found the  feedback to be a bit terse.

**Comment by metasean** [comment]
> Feedback was helpful in general but sometimes useless when missing something tedious...

Too true!

There's definitely a challenge when developing tools like this.  Once you're out of the web-based tutorials, and you're working in an Integrated Development Environment (IDE), there are lots of plug-ins that can help automate tasks and give you more feedback.

Some require some time investment to get up and running to your tasks, but there are quite a few that will help you with the tedious error problems.

**Comment by kassyoveson** [comment]
Thank heavens for IDE's. Visual Code Studio helped integrate plug-ins and it made a world of difference. My main problem at this point of the course was I kept pushing the space bar. (Do you know how hard it is to find an error because of a space? ugggh!) I also liked to add ;;;; where they didn't belong.
